python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -564,7 +565,11 @@ def _fsproxy_wrap(cls, value, func):
         _FinalProxy subclasses can override this classmethod if they
         need particular behaviour when wrapped up.
         """
-        proxy = object.__new__(cls)
+        base_class = _get_proxy_base_class(cls)
+        if base_class is object:
+            proxy = base_class.__new__(cls)
+        else:
+            proxy = base_class.__new__(cls, value)
```

Comment:
Can we explicitly match the type rather than a blanket else statement, or write a helper singledispatch function to accomplish the same (not sure if singledispatch will work with proxies though)? As soon as a new type is added this will fail in a potentially confusing way depending on how the arguments are handled.

Reply:
The singledispatch idea is interesting, I'll give it a try. But will likely go with your first suggestion